File: regions/system/library/swift/session-client/Sources/K1s0SessionClient/Session.swift
import Foundation

public struct Session: Sendable {
    public let id: String
    public let userId: String
    public let token: String
    public let expiresAt: Date
    public let createdAt: Date
    public let revoked: Bool
    public let metadata: [String: String]

    public init(
        id: String,
        userId: String,
        token: String,
        expiresAt: Date,
        createdAt: Date,
        revoked: Bool = false,
        metadata: [String: String] = [:]
    ) {
        self.id = id
        self.userId = userId
        self.token = token
        self.expiresAt = expiresAt
        self.createdAt = createdAt
        self.revoked = revoked
        self.metadata = metadata
    }
}

public struct CreateSessionRequest: Sendable {
    public let userId: String
    public let ttlSeconds: Int
    public let metadata: [String: String]

    public init(userId: String, ttlSeconds: Int, metadata: [String: String] = [:]) {
        self.userId = userId
        self.ttlSeconds = ttlSeconds
        self.metadata = metadata
    }
}

public struct RefreshSessionRequest: Sendable {
    public let id: String
    public let ttlSeconds: Int

    public init(id: String, ttlSeconds: Int) {
        self.id = id
        self.ttlSeconds = ttlSeconds
    }
}

public enum SessionError: Error, Sendable {
    case sessionNotFound(id: String)
    case sessionRevoked(id: String)
    case sessionExpired(id: String)
}

File: regions/system/library/dart/session_client/lib/src/session.dart
class Session {
  final String id;
  final String userId;
  final String token;
  final DateTime expiresAt;
  final DateTime createdAt;
  final bool revoked;
  final Map<String, String> metadata;

  const Session({
    required this.id,
    required this.userId,
    required this.token,
    required this.expiresAt,
    required this.createdAt,
    this.revoked = false,
    this.metadata = const {},
  });

  Session copyWith({
    String? id,
    String? userId,
    String? token,
    DateTime? expiresAt,
    DateTime? createdAt,
    bool? revoked,
    Map<String, String>? metadata,
  }) {
    return Session(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      token: token ?? this.token,
      expiresAt: expiresAt ?? this.expiresAt,
      createdAt: createdAt ?? this.createdAt,
      revoked: revoked ?? this.revoked,
      metadata: metadata ?? this.metadata,
    );
  }
}

class CreateSessionRequest {
  final String userId;
  final int ttlSeconds;
  final Map<String, String>? metadata;

  const CreateSessionRequest({
    required this.userId,
    required this.ttlSeconds,
    this.metadata,
  });
}

class RefreshSessionRequest {
  final String id;
  final int ttlSeconds;

  const RefreshSessionRequest({
    required this.id,
    required this.ttlSeconds,
  });
}
